python/tests/unit/arcticdb/version_store/test_sort_merge.py

```diff
@@ -176,17 +177,26 @@ def test_appended_values_are_after(self, lmdb_library):
         expected_df = pd.DataFrame({"col": range(1, 9)}, index=expected_index)
         assert_frame_equal(lib.read("sym").data, expected_df)
 
-    @pytest.mark.xfail(reason="Does not throw and creates dataframe with unordered index.")
     def test_appended_df_interleaves_with_storage(self, lmdb_library):
         lib = lmdb_library
         initial_df = pd.DataFrame({"col": [1, 3]}, index=pd.DatetimeIndex([np.datetime64('2023-01-01'), np.datetime64('2023-01-03')], dtype="datetime64[ns]"))
         lib.write("sym", initial_df)
         df1 = pd.DataFrame({"col": [2]}, index=pd.DatetimeIndex([np.datetime64('2023-01-02')], dtype="datetime64[ns]"))
         lib.write("sym", df1, staged=True)
-        with pytest.raises(Exception) as exception_info:
+        with pytest.raises(UserInputException) as exception_info:
             lib.sort_and_finalize_staged_data("sym", mode=StagedDataFinalizeMethod.APPEND)
         assert "append" in str(exception_info.value)
 
+    def test_appended_df_start_same_as_df_end(self, lmdb_library):
```

Comment:
This test would be better with multiple rows in both the existing and appended data, and different values in the column